atc/policy/checker.go

```diff
@@ -0,0 +1,144 @@
+package policy
+
+import (
+	"fmt"
+	"strings"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/jessevdk/go-flags"
+)
+
+const ActionUsingImage = "UsingImage"
+
+type PolicyCheckNotPass struct{}
+
+func (e PolicyCheckNotPass) Error() string {
+	return "policy check rejected"
+}
+
+type Filter struct {
+	HttpMethods   []string `long:"policy-check-filter-http-method" description:"API http method to go through policy check"`
+	Actions       []string `long:"policy-check-filter-action" description:"Actions in the list will go through policy check"`
+	ActionsToSkip []string `long:"policy-check-filter-action-skip" default:"UsingImage" description:"Actions the list will not go through policy check"`
```

Comment:
I just noticed this default value here and found it kind of surprising. Can this be made opt-in instead, so there are no defaults here at all?

The concern I have is that I may not care about `UseImage` at all, and then list some HTTP actions to skip, not realizing that I've removed this default value along the way. Doing so may then cause all image usage checks to fail.

Reply:
This is a good point. I have removed `UseImage` from default of `ActionsToSkip`, and I changed the way, only if `UseImage` is defined in `Actions` list, it will go through policy check. So that, users must explicitly define `UseImage` in `Actions` if they want to check `UseImage`.

Reply:
Now the behavior is, if a user only configures a OPA server, but without configuring `HttpMethods` and `Actions`, then no action will go through policy check. Meaning that, by default, no action will go through policy check. Users must explicitly configure which Http methods or actions should go through policy check.